src/test/groovy/sprouts/Properties_List_Spec.groovy

```diff
@@ -973,4 +973,311 @@ class Properties_List_Spec extends Specification
+            change.vals() == Vals.of("d", "k", "o", "k")
+        when : 'We remove values from the list based on a predicate using the `popIfItem` method.'
+            vars.popIfItem(v -> "d" == v || "k" == v)
+        then : 'The `oldValues` of the change delegate should be a property list with the removed properties.'
+            change.oldValues().size() == 3
+            change.oldValues() == Vals.of("d", "k", "k")
+        and : 'The `newValues` of the change should be an empty property list.'
+            change.newValues().isEmpty()
+        and : 'The `index` of the change is `-1`.'
+            change.index() == -1
+        and : 'The `vals` should not contain the removed properties'
+            change.vals() == Vals.of("o")
+    }
+
+    def 'The change delegate contains information about changes made to a "Vars" list by removing a sequence of properties.'() {
+        reportInfo """    
+            When you remove a sequence of properties from a "Vars" list, the change delegate contains information about
+            the change. The `oldValues` of the delegate contains the removed properties, and the `newValues` is always
+            an empty list.
+        """
+        given : 'A "Vars" instance having a few properties.'
+            var vars = Vars.of("a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z")
+        and : 'We register a listener that will record the last change for us.'
+            var change = null
+            vars.onChange({ change = it })
+        when : 'We remove a sequence of values from the list with the `removeFirst` method.'
+            vars.removeFirst(4)
+        then : 'The `oldValues` of the change delegate should be a property list with the removed properties.'
+            change.oldValues().size() == 4
+            change.oldValues() == Vals.of("a", "b", "c", "d")
+        and : 'The `newValues` of the change should be an empty property list.'
+            change.newValues().isEmpty()
+        and : 'The `index` of the change is the index of the first property removed.'
+            change.index() == 0
+        and : 'The `vals` should not contain the removed properties'
+            change.vals() == Vars.of("e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z")
+        when : 'We remove a sequence of values from the list with the `removeLast` method.'
+            vars.removeLast(5)
+        then : 'The `oldValues` of the change delegate should be a property list with the removed properties.'
+            change.oldValues().size() == 5
+            change.oldValues() == Vals.of("v", "w", "x", "y", "z")
+        and : 'The `newValues` of the change should be an empty property list.'
+            change.newValues().isEmpty()
+        and : 'The `index` of the change is the index of the first property removed.'
+            change.index() == 17
```

Comment:
I believe the index should be 12, because that is where we started removing 5 elements

Reply:
The `vars` before removing: 
`[e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z]`
and the 12th element would be `q` and we remove from `v` to `z` and `v` is the 17th element.

Reply:
The purpose of the `index()` is so that a user can maintain 
their own data structure side by side to the list.
So when the user receives the event, what they want is the index at which they
can start adding/removing elements on their data structure.
This is because, although our property list is already up-to-date,
the data structure of the user needs the index telling them at what location
the should add remove the `newValues()`/`oldValues()`.
**So the index is useful if it revers to some place in the already modified property list.**

Reply:
Ok, sorry, I just looked at the code and entered it with the debugger
and you seem to be correct.
So it looks like `subList.clear();`
actually clears the range from which the slice was take in the original JDK `List`?

Reply:
Yes. It was already used in the `popLast` method, so I decided to use it here as well.
And I didn't expect it at first either, but it's exactly what it's supposed to be used for: [List.subList](https://docs.oracle.com/javase/8/docs/api/java/util/List.html#subList-int-int-)